spec/models/ability_spec.rb

```diff
@@ -0,0 +1,121 @@
+require 'spec_helper'
+require 'cancan/matchers'
+
+describe Ability do
+  let(:subject) { described_class.new(user) }
+  let(:user) { mock_user }
+  let(:item) { Dor::Item.new(pid: 'x') }
+
+  context 'as an administrator' do
+    let(:user) { mock_user(is_admin?: true)}
+
+    it { should be_able_to(:manage, :everything) }
+  end
+
+  context 'as a manager' do
+    let(:user) { mock_user(is_manager?: true)}
+
+    it { should_not be_able_to(:manage, :everything) }
+    it { should be_able_to(:manage_item, item) }
+    it { should be_able_to(:manage_content, item) }
+    it { should be_able_to(:manage_desc_metadata, item) }
+    it { should be_able_to(:create, Dor::AdminPolicyObject) }
+  end
+
+  context 'as a viewer' do
+    let(:user) { mock_user(is_viewer?: true)}
+
+    it { should_not be_able_to(:manage_item, item) }
+    it { should_not be_able_to(:manage_content, item) }
+    it { should_not be_able_to(:manage_desc_metadata, item) }
+    it { should be_able_to(:view_item, item) }
+    it { should be_able_to(:view_metadata, item) }
+    it { should be_able_to(:view_content, item) }
+  end
+
+  context 'for items without an APO' do
+    it { should_not be_able_to(:manage_item, item) }
+    it { should_not be_able_to(:manage_content, item) }
+    it { should_not be_able_to(:manage_desc_metadata, item) }
+    it { should_not be_able_to(:view_item, item) }
+    it { should_not be_able_to(:view_content, item) }
+    it { should_not be_able_to(:view_metadata, item) }
+  end
+
+  context 'from an APO' do
+    let(:item) { Dor::AdminPolicyObject.new(pid: 'apo') }
+
+    context 'for a managed item' do
+      before { allow(item).to receive(:can_manage_item?).and_return(true) }
+      it { should be_able_to(:manage_item, item) }
+    end
+
+    context 'for a viewable item' do
+      before { allow(item).to receive(:can_view_item?).and_return(true) }
+      it { should be_able_to(:view_item, item) }
+    end
+  end
+
+  context 'with roles assigned by an APO' do
+    let(:apo) { double(Dor::AdminPolicyObject, pid: 'apo')}
+    let(:random_item) { Dor::Item.new(pid: 'x') }
+
+    before do
+      allow(item).to receive(:admin_policy_object).and_return(apo)
```

Comment:
It would help to add copious commentary on the contexts to explain what they intend to do.  In this particular section, does `random_item` have any APO?  Should it?  Why?  Does the instance get a default APO?  The code alone cannot explain these things and we shouldn't have to be an expert in DOR to understand this spec.